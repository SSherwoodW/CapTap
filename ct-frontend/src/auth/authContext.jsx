import { createContext, useContext, useState, useEffect } from 'react';
import useLocalStorage from '../hooks/useLocalStorage';
import CapTapApi from '../../../api';
import * as jose from 'jose';


export const AuthContext = createContext();

export const useAuth = () => useContext(AuthContext);

export const AuthProvider = ({ children }) => {
    const [isLoggedIn, setIsLoggedIn] = useState();
    const [currentUser, setCurrentUser] = useLocalStorage("currentUser", null);
    const [token, setToken] = useLocalStorage("token", null);

    useEffect(() => {
        async function fetchCurrentUser() {
            try {
                if (token !== null) {
                    CapTapApi.token = token;
                    const { username } = jose.decodeJwt(token);
                    const currentUser = await CapTapApi.getCurrentUser(username);
                    setCurrentUser(currentUser);
                    setIsLoggedIn(true);
                }
            } catch (error) {
                console.error('Error fetching current user:', error);
                setCurrentUser(null);
                setIsLoggedIn(false);
            }
        }

        fetchCurrentUser();
    }, [token]);

    async function handleLogin(formData) {
        try {
            const logInResponse = await CapTapApi.login(formData);
            setToken(logInResponse);
            setIsLoggedIn(true);

            // Fetch user data and set currentUser state
            const { username } = jose.decodeJwt(token);
            const userData = await CapTapApi.getCurrentUser(username);
            setCurrentUser(userData);
            return { success: true };
        } catch (err) {
            console.log("login failed", err)
            return {success: false, error: err.message};
        }
    };

    async function handleSignup(formData) {
        try {
            const signUpResponse = await CapTapApi.signup(formData);
            setToken(signUpResponse);
            localStorage.setItem('token', signUpResponse);
            setIsLoggedIn(true);

            // Fetch user data and set currentUser state
            const { username } = jose.decodeJwt(token);
            const userData = await CapTapApi.getCurrentUser(username);
            setCurrentUser(userData);
            return { success: true };
        } catch (err) {
            console.error("signup failed", err);
            return { success: false, error: err.message };
        }
    };

    const handleLogout = () => {
        localStorage.removeItem('token');
        setToken(null);
        setIsLoggedIn(false);
        setCurrentUser(null);
    };

    return (
        <AuthContext.Provider value={{ isLoggedIn, currentUser, handleLogin, handleSignup, handleLogout }}>
            {children}
        </AuthContext.Provider>
    );
}

export default AuthProvider;