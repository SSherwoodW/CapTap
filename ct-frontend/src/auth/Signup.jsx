import React, { useContext, useState } from "react";
import { Navigate } from "react-router-dom";
import { AuthContext } from "./authContext";

function Signup() {
    const { currentUser, handleSignup } = useContext(AuthContext);
    const [formData, setFormData] = useState({
        firstName: "",
        lastName: "",
        email: "",
        username: "",
        birthdate: "",
        password: "",
    });
    const [formErrors, setFormErrors] = useState([]);

    console.debug(
        "SignupForm",
        "signup=", typeof handleSignup,
        "formData=", formData,
        "formErrors", formErrors,
    );

    /** Handle form submit:
     *
     * Calls signup func prop and, if successful, redirect to /.
     */

    async function handleSubmit(evt) {
        evt.preventDefault();
        console.log("formData in handleSubmit", formData);
        let result = await handleSignup(formData);
        if (result.success) {
        console.log('success in handleSubmit Signup')
        } else {
        setFormErrors(result.errors);
        }
    };

    const handleChange = (evt) => {
        const { name, value } = evt.target;
        setFormData(data => ({
            ...data,
            [name]: value
        }));
    };
    return <>
        <div className="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8">
            {currentUser && (
                <Navigate to="/" replace={true} />
            )}
            <div className="sm:mx-auto sm:w-full sm:max-w-sm">
                <img className="mx-auto h-10 w-auto" src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600" alt="CapTap"></img>
                <h2 className="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900">Sign up for your account</h2>
            </div>

            <div className="mt-10 sm:mx-auto sm:w-full sm:max-w-sm">
                <form className="space-y-4" onSubmit={handleSubmit}>
                    <div className="flex items-center justify-between">
                        <div>
                            <label htmlFor="firstName" className="block text-sm font-medium leading-6 text-gray-900">First Name</label>
                            <div className="mt-2">
                                <input id="firstName" name="firstName" type="text" value={formData.firstName} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                            </div>
                        </div>
                        <div>
                            <label htmlFor="lastName" className="block text-sm font-medium leading-6 text-gray-900">Last Name</label>
                            <div className="mt-2">
                                <input id="lastName" name="lastName" type="text" value={formData.lastName} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                            </div>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="birthdate" className="block text-sm font-medium leading-6 text-gray-900">Birthdate</label>
                        <div className="mt-2">
                            <input id="birthdate" name="birthdate" type="date" value={formData.birthdate} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="email" className="block text-sm font-medium leading-6 text-gray-900">Email</label>
                        <div className="mt-2">
                            <input id="email" placeholder="ex: you@captap.com" name="email" type="email" value={formData.email} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>
                    <div>
                        <label htmlFor="username" className="block text-sm font-medium leading-6 text-gray-900">Username</label>
                        <div className="mt-2">
                            <input id="username" autoComplete="username" name="username" type="text" value={formData.username} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <div className="flex items-center justify-between">
                            <label htmlFor="password" className="block text-sm font-medium leading-6 text-gray-900">Password</label>
                            <div className="text-sm">
                                <a href="#" className="font-semibold text-indigo-600 hover:text-indigo-500">Forgot password?</a>
                            </div>
                        </div>
                        <div className="mt-2">
                            <input id="password" autoComplete="current-password" name="password" type="password" value={formData.password} onChange={handleChange} required className="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"/>
                        </div>
                    </div>

                    <div>
                        <button type="submit" className="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600">Sign up</button>
                    </div>
                </form>

                <p className="mt-10 text-center text-sm text-gray-500">
                Already registered?
                    <a href="/login" className="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"> Log In</a>
                </p>
            </div>
        </div>
    </>
}

export default Signup;