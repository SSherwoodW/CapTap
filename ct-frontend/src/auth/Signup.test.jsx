import React from "react";
import { render, screen } from '@testing-library/react';
import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest';
import { AuthContext } from '../auth/authContext'; 

import Signup from "./Signup";

vi.mock('react-router-dom', async () => {
    const mod = await vi.importActual('react-router-dom');
    return {
        ...mod,
        useLocation: () => ({
            location: {pathname: "/"}
        }),
        Navigate: ({ to }) => <div data-testid="navigate">{to}</div>
    }
})

describe('Signup', () => {
    beforeEach(() => {
        const mockAuthContext = {
        currentUser: { username: 'testuser' } 
        };
        vi.spyOn(React, 'useContext').mockReturnValue(mockAuthContext);
    });

    afterEach(() => {
        vi.restoreAllMocks();
    });
    
    it('renders the Signup component', () => {
        render(
            <AuthContext.Provider value={{ currentUser: { username: 'testuser' } }}>
                <Signup />
            </AuthContext.Provider>
        )
        screen.debug();
    });

    it('matches snapshot', () => {
        const signupPage = render(
            <AuthContext.Provider value={{ currentUser: { username: 'testuser' } }}>
                <Signup />
            </AuthContext.Provider>
        )
        screen.debug();
        expect(signupPage).toMatchSnapshot();
    });

    it('matches snapshot when logged out', () => {
        const signupPage = render(
            <AuthContext.Provider value={{ currentUser: null }}>
                <Signup />
            </AuthContext.Provider>
        )
        screen.debug();
        expect(signupPage).toMatchSnapshot();
    });
})