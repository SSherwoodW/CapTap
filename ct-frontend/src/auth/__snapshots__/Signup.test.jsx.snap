// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Signup > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8"
      >
        <div
          data-testid="navigate"
        >
          /
        </div>
        <div
          class="sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <img
            alt="CapTap"
            class="mx-auto h-10 w-auto"
            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
          />
          <h2
            class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
          >
            Sign up for your account
          </h2>
        </div>
        <div
          class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <form
            class="space-y-4"
          >
            <div
              class="flex items-center justify-between"
            >
              <div>
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="firstName"
                >
                  First Name
                </label>
                <div
                  class="mt-2"
                >
                  <input
                    class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                    id="firstName"
                    name="firstName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div>
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="lastName"
                >
                  Last Name
                </label>
                <div
                  class="mt-2"
                >
                  <input
                    class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                    id="lastName"
                    name="lastName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="birthdate"
              >
                Birthdate
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="birthdate"
                  name="birthdate"
                  required=""
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="email"
              >
                Email
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="email"
                  name="email"
                  placeholder="ex: you@captap.com"
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="username"
              >
                Username
              </label>
              <div
                class="mt-2"
              >
                <input
                  autocomplete="username"
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <div
                class="flex items-center justify-between"
              >
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="text-sm"
                >
                  <a
                    class="font-semibold text-indigo-600 hover:text-indigo-500"
                    href="#"
                  >
                    Forgot password?
                  </a>
                </div>
              </div>
              <div
                class="mt-2"
              >
                <input
                  autocomplete="current-password"
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                type="submit"
              >
                Sign up
              </button>
            </div>
          </form>
          <p
            class="mt-10 text-center text-sm text-gray-500"
          >
            Already registered?
            <a
              class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
              href="/login"
            >
               Log In
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8"
    >
      <div
        data-testid="navigate"
      >
        /
      </div>
      <div
        class="sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <img
          alt="CapTap"
          class="mx-auto h-10 w-auto"
          src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
        />
        <h2
          class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
        >
          Sign up for your account
        </h2>
      </div>
      <div
        class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <form
          class="space-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="firstName"
              >
                First Name
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="firstName"
                  name="firstName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="lastName"
              >
                Last Name
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="lastName"
                  name="lastName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="birthdate"
            >
              Birthdate
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="birthdate"
                name="birthdate"
                required=""
                type="date"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="email"
            >
              Email
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="email"
                name="email"
                placeholder="ex: you@captap.com"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="username"
            >
              Username
            </label>
            <div
              class="mt-2"
            >
              <input
                autocomplete="username"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="username"
                name="username"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <div
              class="flex items-center justify-between"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="password"
              >
                Password
              </label>
              <div
                class="text-sm"
              >
                <a
                  class="font-semibold text-indigo-600 hover:text-indigo-500"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
            </div>
            <div
              class="mt-2"
            >
              <input
                autocomplete="current-password"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              type="submit"
            >
              Sign up
            </button>
          </div>
        </form>
        <p
          class="mt-10 text-center text-sm text-gray-500"
        >
          Already registered?
          <a
            class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
            href="/login"
          >
             Log In
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Signup > matches snapshot when logged out 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8"
      >
        <div
          class="sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <img
            alt="CapTap"
            class="mx-auto h-10 w-auto"
            src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
          />
          <h2
            class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
          >
            Sign up for your account
          </h2>
        </div>
        <div
          class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
        >
          <form
            class="space-y-4"
          >
            <div
              class="flex items-center justify-between"
            >
              <div>
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="firstName"
                >
                  First Name
                </label>
                <div
                  class="mt-2"
                >
                  <input
                    class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                    id="firstName"
                    name="firstName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div>
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="lastName"
                >
                  Last Name
                </label>
                <div
                  class="mt-2"
                >
                  <input
                    class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                    id="lastName"
                    name="lastName"
                    required=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="birthdate"
              >
                Birthdate
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="birthdate"
                  name="birthdate"
                  required=""
                  type="date"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="email"
              >
                Email
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="email"
                  name="email"
                  placeholder="ex: you@captap.com"
                  required=""
                  type="email"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="username"
              >
                Username
              </label>
              <div
                class="mt-2"
              >
                <input
                  autocomplete="username"
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <div
                class="flex items-center justify-between"
              >
                <label
                  class="block text-sm font-medium leading-6 text-gray-900"
                  for="password"
                >
                  Password
                </label>
                <div
                  class="text-sm"
                >
                  <a
                    class="font-semibold text-indigo-600 hover:text-indigo-500"
                    href="#"
                  >
                    Forgot password?
                  </a>
                </div>
              </div>
              <div
                class="mt-2"
              >
                <input
                  autocomplete="current-password"
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
              </div>
            </div>
            <div>
              <button
                class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
                type="submit"
              >
                Sign up
              </button>
            </div>
          </form>
          <p
            class="mt-10 text-center text-sm text-gray-500"
          >
            Already registered?
            <a
              class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
              href="/login"
            >
               Log In
            </a>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex min-h-full flex-col justify-center px-6 py-12 lg:px-8"
    >
      <div
        class="sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <img
          alt="CapTap"
          class="mx-auto h-10 w-auto"
          src="https://tailwindui.com/img/logos/mark.svg?color=indigo&shade=600"
        />
        <h2
          class="mt-10 text-center text-2xl font-bold leading-9 tracking-tight text-gray-900"
        >
          Sign up for your account
        </h2>
      </div>
      <div
        class="mt-10 sm:mx-auto sm:w-full sm:max-w-sm"
      >
        <form
          class="space-y-4"
        >
          <div
            class="flex items-center justify-between"
          >
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="firstName"
              >
                First Name
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="firstName"
                  name="firstName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div>
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="lastName"
              >
                Last Name
              </label>
              <div
                class="mt-2"
              >
                <input
                  class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                  id="lastName"
                  name="lastName"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="birthdate"
            >
              Birthdate
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="birthdate"
                name="birthdate"
                required=""
                type="date"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="email"
            >
              Email
            </label>
            <div
              class="mt-2"
            >
              <input
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="email"
                name="email"
                placeholder="ex: you@captap.com"
                required=""
                type="email"
                value=""
              />
            </div>
          </div>
          <div>
            <label
              class="block text-sm font-medium leading-6 text-gray-900"
              for="username"
            >
              Username
            </label>
            <div
              class="mt-2"
            >
              <input
                autocomplete="username"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="username"
                name="username"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
          <div>
            <div
              class="flex items-center justify-between"
            >
              <label
                class="block text-sm font-medium leading-6 text-gray-900"
                for="password"
              >
                Password
              </label>
              <div
                class="text-sm"
              >
                <a
                  class="font-semibold text-indigo-600 hover:text-indigo-500"
                  href="#"
                >
                  Forgot password?
                </a>
              </div>
            </div>
            <div
              class="mt-2"
            >
              <input
                autocomplete="current-password"
                class="block w-full rounded-md border-0 py-1.5 text-gray-900 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6"
                id="password"
                name="password"
                required=""
                type="password"
                value=""
              />
            </div>
          </div>
          <div>
            <button
              class="flex w-full justify-center rounded-md bg-indigo-600 px-3 py-1.5 text-sm font-semibold leading-6 text-white shadow-sm hover:bg-indigo-500 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-indigo-600"
              type="submit"
            >
              Sign up
            </button>
          </div>
        </form>
        <p
          class="mt-10 text-center text-sm text-gray-500"
        >
          Already registered?
          <a
            class="font-semibold leading-6 text-indigo-600 hover:text-indigo-500"
            href="/login"
          >
             Log In
          </a>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
